pennylane/transforms/qcut.py

```diff
@@ -428,3 +429,320 @@ def contract_tensors(
     kwargs = {} if use_opt_einsum else {"like": tensors[0]}
 
     return contract(eqn, *tensors, **kwargs)
+
+
+@dataclass()
+class CutStrategy:
+    """
+    A circuit-cutting distribution policy for executing (large) circuits on available (comparably
+    smaller) devices.
+
+    Args:
+        devices (Union[qml.Device, Sequence[qml.Device]]): Single, or Sequence of, device(s).
+            Optional only when ``max_free_wires`` is provided.
+        max_free_wires (int): Number of wires for the largest available device. Optional only when
+            ``devices`` is provided where it defaults to the maximum number of wires among
+            ``devices``.
+        min_free_wires (int): Number of wires for the smallest available device.
+            Optional, defaults to ``max_device_wires``.
+        max_fragments_probed (Union[int, Sequence[int]]): Single, or 2-Sequence of, number(s)
+            specifying the potential (range of) number of fragments for the partitioner to attampt.
+            Optional, defaults to probing all valid strategies derivable from the circuit and
+            devices.
+        max_free_gates (int): Maximum allowed circuit depth for the deepest available device.
+            Optional, defaults to unlimited depth.
+        min_free_gates (int): Maximum allowed circuit depth for the shallowest available device.
+            Optional, defaults to ``max_device_gates``.
+
+    **Example**
+
+    .. code-block:: python
+
+        import pennylane as qml
+        from pennylane.transforms import qcut
+
+        dev_a = qml.device('default.qubit', wires=4)
+        dev_b = qml.device('default.qubit', wires=6)
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(0.4, wires=0)
+            qml.RY(0.9, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.expval(qml.PauliZ(1))
+
+        tape_dag = qcut.tape_to_graph(tape)
+
+    Cut circuit with single-device-based strategy using the default 'kahypar' cutter:
+    >>> cut_strategy = qcut.CutStrategy(devices=dev_a)
+    >>> qcut.cut_circuit(tape_dag, method='kahypar', strategy=cut_strategy)
+
+    Cut circuit with multi-device-based strategy using the default 'kahypar' cutter:
+    >>> cut_strategy = qcut.CutStrategy(devices=(dev_a, dev_b))
+    >>> qcut.cut_circuit(tape_dag, method='kahypar', strategy=cut_strategy)
+
+    Cut circuit with user-supplied strategy using user-supplied partitioning callable:
+    >>> cut_strategy = qcut.CutStrategy(
+            max_free_wires=6,
+            min_free_wires=4,
+            num_fragments_probed=(2, 5),
+        )
+    >>> qcut.cut_circuit(tape_dag, method=my_partitioner, strategy=cut_strategy, **my_kwargs)
```

Comment:
Nice to see this mapped out! However, since it's blocked by the `cut_circuit` transform, how about this example goes into the "Usage details" section of `cut_circuit` and for the examples in this function we focus just on a simple use case of `CutStrategy`, e.g.,

```rst
    The following cut strategy specifies that a circuit should be cut into between
    ``2`` to ``5`` fragments, with each fragment having at most ``6`` wires and
    at least ``4`` wires:

    >>> cut_strategy = qcut.CutStrategy(
    ...     max_free_wires=6,
    ...     min_free_wires=4,
    ...     num_fragments_probed=(2, 5),
    ... )
```